### PR TITLE
Improves Architecture Documentation

### PR DESCRIPTION
**Pull request by ericbaum**
- Full view of the architecture
- Description of each piece of the architecture
- Tags sessions that have missing content with a **TODO**
- Translates the architecture documentation to portuguese

resolves dojot/dojot#142

### CODE REVIEW

File: source/architecture.rst [hunk 1]
```diff
@@ -67,16 +63,134 @@ converging to a reviewed architectural proposition as described on :numref:`new_
...
+Each one of the components that are part of the architecture are briefly described on the sub-sections below.
+
+Kafka + Subscription Manager + NGSI
+***********************************
+
+Using Apache Kafka in conjunction with a subscriptions managers and a NGSI interface translator we compose a context
+broker analogous to Fiware Orion, with the addition of functionalities that are tipical to brokers, more processing
+power and scalability. Apache Kafka is a distributed platform that was consolidated for the construction of data
+pipelines in real time and streaming applications, differently from the Fiware Orion core that is a non-relational DB.
+
```
Comment (giovannicuriel) [inline source/architecture.rst]: I would change this paragraph to:
"Apache Kafka is a distributed messaging platform that can be used by applications which need to stream data or consume/produce data pipelines. In contrast to what Orion is, Kafka seems to be more appropriate to fulfil dojot's architectural requirements (responsibility isolation, simplicity, and so on). And using it with a subscription manager and a NGSI interface translator, we can compose an entity which is very close to the features offered by Orion, in addition to improved speed and easier scalability."

File: source/architecture.rst [hunk 2]
```diff
@@ -19,6 +19,61 @@ This section should describe - in high level - the components that comprise the
...
+.. _old_arch:
+.. figure:: images/old_architecture.png
+    :width: 100%
+    :align: center
+    :alt: Initial Architecture based on the Fiware project
+
+    Initial Architecture
+
+In this proposal we utilize the following components from Fiware: the Orion context broker as the communication bus for
```
[the remark this answers is not in the document]
Comment (ericbaum) [reply source/architecture.rst]: Ok, will try this style

File: source/architecture.rst [hunk 3]
```diff
@@ -19,6 +19,61 @@ This section should describe - in high level - the components that comprise the
...
+all the internal services of the middleware, the history service, STH, for storing IoT devices data in a non-relational
+MongoDB database, the Perseo CEP service that is responsible for treating events in real time and the iot-agent service
+as an abstraction layer for integrating MQTT and HTTP devices.
+
+To this services we added the Kong API Gateway to act as a centralized point-of-access removing the need of direct
```
[the remark this answers is not in the document]
Comment (ericbaum) [reply source/architecture.rst]: Done!

File: source/architecture.rst (hunk 1 above)
Comment (ericbaum) [reply source/architecture.rst]: Yeah, much better

File: source/architecture.rst [hunk 4]
```diff
@@ -19,6 +19,61 @@ This section should describe - in high level - the components that comprise the
...
+GUI or directly using the REST APIs provided by the API Gateway, he also configures processing flows to deal with the
+data generated by the configured devices. As an example we can generate notifications when the data of a device has one
+of its values reaching a threshold or we can add an entry to a database when a device enters or leaves an specific
+geographic location. This user operations on the API result in configurations across the internal services of the
+middleware (Broker, CEP and iot-agent), being partially abstracted by the orchestration service.
+
+The user contexts are isolated and there is no data sharing, the access credentials are validated by the authorization
+service for each and every operation (API Request). Once the devices and the flows are configured, the iot-agent is
+capable of mapping the data received from devices, encapsulated on MQTT for example, and send then to the context broker
+for internal distribution, reaching, for instance, the historic service so it can persist the data on a database and the
```
[the remark this answers is not in the document]
Comment (ericbaum) [reply source/architecture.rst]: I guess "history" would be  the correct term.

**Comment by ericbaum** [comment]
Requested changes were added, waiting for a new review

**Pull request merged**